File: src/resources/scorers.ts
import type { HttpClient } from '../http.js';
import type { Scorer, CreateScorerBody, UpdateScorerBody } from '../types/scorer.js';

export class ScorersResource {
  constructor(private http: HttpClient) {}

  async list(): Promise<Scorer[]> {
    return this.http.get<Scorer[]>('/v1/scorers');
  }

  async get(id: string): Promise<Scorer> {
    return this.http.get<Scorer>(`/v1/scorers/${id}`);
  }

  async create(body: CreateScorerBody): Promise<Scorer> {
    return this.http.post<Scorer>('/v1/scorers', body);
  }

  async update(id: string, body: UpdateScorerBody): Promise<Scorer> {
    return this.http.patch<Scorer>(`/v1/scorers/${id}`, body);
  }

  async delete(id: string): Promise<{ ok: boolean }> {
    return this.http.delete<{ ok: boolean }>(`/v1/scorers/${id}`);
  }
}

File: packages/python/invariance/resources/scorers.py
from __future__ import annotations

from typing import Any

from ..http_client import HttpClient


class ScorersResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def list(self) -> list[dict[str, Any]]:
        return await self._http.get("/v1/scorers")

    async def get(self, id: str) -> dict[str, Any]:
        return await self._http.get(f"/v1/scorers/{id}")

    async def create(self, body: dict[str, Any]) -> dict[str, Any]:
        return await self._http.post("/v1/scorers", body)

    async def update(self, id: str, body: dict[str, Any]) -> dict[str, Any]:
        return await self._http.patch(f"/v1/scorers/{id}", body)

    async def delete(self, id: str) -> dict[str, Any]:
        return await self._http.delete(f"/v1/scorers/{id}")
